crypto/store/store_lib.c

```diff
@@ -1095,3 +1095,66 @@ OSSL_STORE_CTX *OSSL_STORE_attach(BIO *bp, const char *scheme,
 
     return ctx;
 }
+
+const char *OSSL_file_stat(const char *uri, struct stat *st)
+{
+    const char *path = uri, *q;
+    struct stat local_st;
+
+    if (st == NULL)
+        st = &local_st;
+
+    /*
+     * First, unless the URI starts with "file://",
+     * try and see if the full URI can be taken as a local file path name.
+     */
+    if (!HAS_CASE_PREFIX(uri, "file://")) {
+        if (stat(path, st) == 0)
+            return uri;
+        ERR_raise_data(ERR_LIB_SYS, errno, "calling stat(%s)", path);
+    }
```

Comment:
Nit: the mix of variables here is a bit strange...

Reply:
Well, I put `path` here on purpose, because when the guard is fulfilled, we know we have a path to try out.